lochness/rpms/__init__.py

```diff
@@ -221,24 +221,29 @@ def sync(Lochness, subject, dry=False):
                                BIDS=Lochness['BIDS'])
         proc_dst = Path(proc_folder) / f"{subject_id}_{measure}.csv"
 
-        # load the time of the lastest data pull from daris
-        # estimated from the mtime of the zip file downloaded
+        # if the csv already exists, compare the dataframe
         if Path(target_df_loc).is_file():
-            latest_pull_mtime = target_df_loc.stat().st_mtime
+            # index might be different, so drop it before comparing it
+            prev_df = pd.read_csv(target_df_loc).reset_index().drop(
+                    'index', axis=1)
+
+            # in order to use df.equals function, which also checks for data
+            # types of each data, the source_df needs to be saved and re-loaded
+            # to make the datatype consistent to that of prev_df
+            with tf.NamedTemporaryFile(delete=True) as f:
+                source_df.to_csv(f.name, index=False)
+                same_df = pd.read_csv(f.name).reset_index().drop(
+                        'index', axis=1).equals(prev_df)
```

Comment:
Equivalent to `.reset_index(inplace=True, drop=True)`

Reply:
I'll update to your suggestion in the next PR

Reply:
@tashrifbillah it was not equivalent to `.reset_index(inplace=True, drop=True)`

including `inplace=True` results in `None` returned, where we need it to return `pd.DataFrame` to apply `.equals`

Reply:
Okay I get what you say:
```python
In [0]: df= pd.read_csv('files-ProNET-flowcheck-day1to9999.csv'

In [1]: tmp= df.groupby('site').get_group('GA')

In [2]: tmp1= df.groupby('site').get_group('GA')

In [3]: tmp.reset_index(inplace=True, drop=True).equals(tmp1)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-31-70dc6a952c91> in <module>
----> 1 tmp.reset_index(inplace=True, drop=True).equals(tmp1)

AttributeError: 'NoneType' object has no attribute 'equals'
```

In that case, all you have to do is write two lines:

```python
In[3]: tmp.reset_index(inplace=True, drop=True).equals(tmp1)

In [4]: tmp.equals(tmp1)
Out[4]: False
```

---

Bottom line--my equivalence comment was mainly for my understanding. I have zero problem with the way you had done it originally.